src/main/java/org/eclipse/emfcloud/jackson/databind/deser/EObjectDeserializer.java

```diff
@@ -60,7 +61,11 @@ public EObject deserialize(final JsonParser jp, final DeserializationContext ctx
       if (feature == null && defaultType != null) {
          propertyMap = builder.construct(ctxt, defaultType);
       } else if (feature instanceof EReference) {
-         propertyMap = builder.construct(ctxt, ((EReference) feature).getEReferenceType());
+         final EObject parent = EMFContext.getParent(ctxt);
+         final EClass parentClass = parent != null ? parent.eClass() : null;
+         final EGenericType reifiedType = EcoreUtil.getReifiedType(parentClass, feature.getEGenericType());
```

Comment:
Calling `EcoreUtil.getReifiedType` with a null parent will cause a NPE. I think we should check that before calling the method and maybe use the previous behavior as a fallback, what do you think?